### PR TITLE
[CDAP-8900] DataPrep upgrade ability if newer artifact exist

### PR DESCRIPTION
**Pull request by elfenheart**
- If newer artifact exist, Upgrade button should show up
- During service enable, check for newer artifact version. If exist, create new app with newer version
- Use the `plugin.version` specified by wrangler-service when add to pipeline.

JIRA: https://issues.cask.co/browse/CDAP-8900

### CODE REVIEW

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/UpgradeModal.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        isOpen={true}
+        toggle={this.attemptClose}
+        zIndex="1070"
+        className="dataprep-upgrade-modal"
+      >
+        <ModalHeader>
+          <span>
+            Upgrade Data Preparation
+          </span>
+
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/UpgradeModal.js]: unnecessary space?

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/UpgradeModal.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                  className="fa fa-times"
+                  onClick={this.attemptClose}
+                />
+              </div>
+            )
+          }
+        </ModalHeader>
+        <ModalBody>
+          {content}
+
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/UpgradeModal.js]: here too

**Comment by tonybach** [comment]
Minor comments, otherwise code LGTM 👍

**Comment by elfenheart** [comment]
Build: http://builds.cask.co/browse/CDAP-DRC6054-3

**Comment by bdmogal** [comment]
LGTM

**Pull request merged**